File: 1431. Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result = []
        for el in candies:
            for i in candies:
                if el+extraCandies<i:
                    result.append(False)
                    break
            else:
                result.append(True)
        return result

File: 1431. Kids With the Greatest Number of Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> result;
        int n = candies.size();
        int i, j;
        for(i=0; i<n; i++) {
            int flag = 1;
            for (j=0; j<n; j++) {
                if (candies[i] + extraCandies < candies[j]) {
                    result.push_back(false);
                    flag = 0;
                    break;
                }
            }
            if (flag)
                result.push_back(true);
        }
        return result;
    }
};
